src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1572,6 +1574,11 @@ internal void Validate()
                 throw new ArgumentException("Host can't be null");
             if (Multiplexing && !Pooling)
                 throw new ArgumentException("Pooling must be on to use multiplexing");
+            if (SslMode == SslMode.Require && !TrustServerCertificate)
```

Comment:
Just remembered we also said we'll throw if TSC is true and SslMode is Verify*. We can also do the same for Allow as it's new, and maybe even for Disable since it makes no sense...